pkg/imrpc/instance.proto

```diff
@@ -0,0 +1,94 @@
+syntax="proto3";
+
+package imrpc;
+
+import "google/protobuf/empty.proto";
+import "imrpc.proto";
+
+service InstanceService {
+    rpc InstanceCreate(InstanceCreateRequest) returns (InstanceResponse) {}
+	rpc InstanceDelete(InstanceDeleteRequest) returns (InstanceResponse) {}
+	rpc InstanceGet(InstanceGetRequest) returns (InstanceResponse) {}
+	rpc InstanceList(google.protobuf.Empty) returns (InstanceListResponse) {}
+	rpc InstanceLog(InstanceLogRequest) returns (stream LogResponse) {}
+	rpc InstanceWatch(google.protobuf.Empty) returns (stream InstanceResponse) {}
+	rpc InstanceReplace(InstanceReplaceRequest) returns (InstanceResponse) {}
+
+	rpc VersionGet(google.protobuf.Empty) returns(VersionResponse);
+}
+
+message ProcessSpecific {
+	string binary = 1;
+	repeated string args = 2;
+}
+
+message SpdkSpecific {
+	string address = 1;
+    map<string, string> replica_address_map = 2;
+    string frontend = 3;
```

Comment:
NIT: Wrong indent

Reply:
I am thinking if there is a better way to input the necessary args for engines or replicas.

Reply:
Agree. Current layout is a bit messy.

Reply:
Can we add `replica_specific` and `engine_specific` structs? WDYT?

Reply:
Actually, another way to handle this is similar to backing image creation/the process starting: Use a string map to store all necessary input fields. Then there won't be an empty value for fields like `frontend` in replica `InstanceCreate` requests. But the defect is, new developers do not know which fields are necessary by checking the API definition.
I am fine with either. WDYT?

Reply:
How about add the tow maps `replica_specific map[string]string` and `engine_specific map[string]string`?